rfcs/0000-kuadrant_sub_components_configurations.md

```diff
@@ -0,0 +1,248 @@
+The defaults would for the fields are the production recommendations.
+- Reconciled by others, this mostly affects Limitador as the deployment configuration and runtime configuration are in the same CR.
+In the case of Kuadrant the runtime configuration for Limitador is added via the RateLimitingPolicy CR.
+
+### Authorino Spec
+
+#### Unsupported
+- clusterWide
+- authConfigLabelSelectors
+- secrtLabelSelectors
+
+#### Dev/Testing focused
+- image
+- imagePullPolicy
+- logMode
+
+### Limitador Spec
+
+#### Unsupported
+- RateLimitHeaders
+
+#### Reconciled by others
+- Limits
+
+#### Deprecated
+- version
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+As the Kuadrant CR spec will be a sub set of the features that can be configured in the sub components spec, extra maintenances will be required to ensure specs are in sync.
+
+New features of a component will not be accessible in Kuadrant initially.
+This is both a pro and a con.
+
+Documentation becomes harder, as the sub component should be documenting their own features but in Kuadrant the user does not configure the feature in sub component.
+This has the risk of confusing new users.
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+- Why is this design the best in the space of possible designs?
+- What other designs have been considered and what is the rationale for not choosing them?
+- What is the impact of not doing this?
+
+One alternative that was being looked at was allowing the user to bring their own Limitador instances by stating which Limitador CR Kuadrant should use. 
+A major point of issue with this approach was knowing what limits the user had configured and what limits Kuadrant configured. 
+Sharing global counters is a valid reason to want to share Limitador instances. 
+How ever it this case Limitador would not be using one replica and therefore would have a back-end storage configured.
+It is the back-end storage that needs to be shared across instances.
+This can be done with adding the configuration in the Kuadrant CR.
+
+# Prior art
+[prior-art]: #prior-art
+
+Discuss prior art, both the good and the bad, in relation to this proposal.
+A few examples of what this can include are:
+
+- Does another project have a similar feature?
+- What can be learned from it? What's good? What's less optimal?
+- Papers: Are there any published papers or great posts that discuss this? If you have some relevant papers to refer to, this can serve as a more detailed theoretical background.
+
+This section is intended to encourage you as an author to think about the lessons from other tentatives - successful or not, provide readers of your RFC with a fuller picture.
+
+Note that while precedent set by other projects is some motivation, it does not on its own motivate an RFC.
+
+------
+In RHOAM a user can configure the custom SMTP and custom domains for 3scale from the addon forms. 
```

Comment:
Would it be possible to not use RHOAM as an example? I think it is too niche of an example to make sense for most users (e.g. Red Hat external contributors)

Reply:
I don't mind removing this, but I don't have a good alternative example. I did not think of the example from a external contributors point of view.